protocols/v2/noise-sv2/src/aed_cipher.rs

```diff
@@ -1,16 +1,60 @@
+// # AEAD Cipher
+//
+// Defines the [`AeadCipher`] trait which standardizes the interface for Authenticated Encryption
+// with Associated Data (AEAD) ciphers used within the Noise protocol implementation. The trait is
+// implemented for two common ciphers, [`ChaCha20Poly1305`] and [`Aes256Gcm`], providing encryption
+// and decryption functionality with authenticated data.
+//
+// ## Overview
+//
+// AEAD ciphers provide both confidentiality and integrity by encrypting data and creating an
+// authentication tag in a single step. The integrity of additional associated data (AAD) can also
+// be verified without including it in the encrypted output.
+//
+// ## Usage
+//
+// The [`AeadCipher`] trait is used by the [`crate::handshake::HandshakeOp`] trait to perform
+// cryptographic operations during the Noise protocol handshake, ensuring secure communication
+// between parties.
+
 use aes_gcm::Aes256Gcm;
 use chacha20poly1305::{aead::Buffer, AeadInPlace, ChaCha20Poly1305, ChaChaPoly1305, KeyInit};
 
+// Defines the interface for AEAD ciphers.
+//
+// The [`AeadCipher`] trait provides a standard interface for initializing AEAD ciphers, and for
+// performing encryption and decryption operations with additional Authenticated Associated Data (AAD). This trait is implemented
+// by either the [`ChaCha20Poly1305`] or [`Aes256Gcm`] specific cipher types, allowing them to be
+// used interchangeably in cryptographic protocols. It is utilized by the
+// [`crate::handshake::HandshakeOp`] trait to secure the handshake process.
+//
+// The `T: Buffer` represents the data buffer to be encrypted or decrypted. The buffer must
+// implement the [`Buffer`] trait, which provides necessary operations for in-place encryption and
+// decryption.
```

Comment:
This trait is public, so the docs should be `///`. also, IIRC if the trait is public all of its functions are public as well, so they should also be using `///` instead of `//`. If the trait is not exported from the main `noise-sv2`  file, maybe it is worth changing the visibility to `pub(crate)`

Reply:
However, the module is not public and therefore not accessible to users of this crate, and the trait is not relevant for any user, so there is no need to expose it.

Reply:
hmm ok, it is worth changing all the `pub` to `pub(crate)` to prevent this confusion - otherwise we need always to refer back to the `lib` file to see whats exported

Reply:
I think these comments are fine like this, and we don't really need to change visibility for now.

there's some discussion about whether this trait should even continue to exist in the future https://github.com/stratum-mining/stratum/issues/1130#issuecomment-2372447388

Reply:
The comments are the same across different files which are `pub` but they are `pub(crate)` so I think visibility should be changed, probably not in this pr but in the refactor one.

Reply:
https://github.com/stratum-mining/stratum/issues/1130#issuecomment-2408165309